lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+			#{$self}__content {
+				padding-left: $space-m;
+				padding-right: $space-m;
+			}
+		}
+	}
+
+	&__content {
+		padding: $space-l $space-s;
+
+		@media #{breakpoint-s()} {
+			padding: $space-l $space-xl;
+		}
+
+		&.-centered {
+			#{$self}__header,
+			#{$self}__textContent {
+				text-align: center;
+			}
+		}
+	}
+
+	&__header {
+		display: flex;
+		flex-direction: column;
+	}
+
+	&__headerFeatureIcon {
+		align-self: center;
+		margin-bottom: $space-m;
+	}
+
+	&__headerTitle {
+		@include displayHeading-xs-default-bold;
+
+		margin-bottom: $space-xs;
+		margin-top: 0;
+
+		&.-small {
+			@include heading-xl-default-bold;
+		}
+	}
+
+	&__headerSubtitle {
+		@include heading-m-default-regular;
+
+		margin-bottom: $space-s;
+		margin-top: 0;
+	}
+
+	&__textContent {
+		@include text-m-default-regular;
+
+		margin-bottom: $space-s;
+	}
+
+	&__checkbox {
+		cursor: pointer;
+		display: flex;
+		align-items: center;
+	}
+
+	&__checkboxLabel {
+		color: $color-neutral-text-heavy;
+		cursor: pointer;
+		margin-left: $space-xxs;
+
+		/** align vertical middle fix */
+		margin-top: -$space-xxxxxs;
+	}
+
+	&__close {
+		border-radius: 100%;
+		color: $color-neutral-icon-weak;
+		position: absolute;
+		right: $space-xs;
+		top: $space-xs;
+	}
+
+	&__image {
+		height: $image-height;
+		background-size: cover;
+	}
+
+	&__footer {
+		display: flex;
+		justify-content: space-between;
+		flex-direction: column-reverse;
+		padding-top: $space-m;
+
+		@media #{breakpoint-s()} {
+			flex-direction: row;
+		}
+
+		&.-singleColumn {
+			#{$self}__footerColumn {
+				width: 100%;
+				justify-content: center;
+			}
+		}
+	}
+
+	&__footerColumn {
+		display: flex;
+		gap: 0 $space-s;
+
+		&.--cta {
+			flex-direction: column-reverse;
+			margin-bottom: $space-xs;
+
+			@media #{breakpoint-s()} {
+				flex-direction: row;
+				margin-bottom: 0;
+			}
+		}
+
+		&.--ctaSecondary {
+			gap: 0 $space-m;
+			justify-content: center;
+
+			@media #{breakpoint-s()} {
+				justify-content: left;
+			}
+		}
+	}
+
+	&__buttonPrimary {
+		margin-bottom: $space-s;
+
+		@media #{breakpoint-s()} {
+			margin-bottom: 0;
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import FeatureIcon from '../../Icons/FeatureIcon/FeatureIcon.vue';
+import { MODAL_SIZES, MODAL_HEADER_TITLE_SIZES } from './Modal.consts';
+import { VueConstructor } from 'vue';
+import { ICONS, ICON_SIZES } from '../../Icons/Icon';
+import { FEATURE_ICON_COLOR, FEATURE_ICON_SIZES } from '../../Icons/FeatureIcon';
+import WnlButton, { BUTTON_COLORS, BUTTON_TYPES, BUTTON_ELEVATIONS } from '../../Buttons/Button';
+import WnlIconButton, { ICON_BUTTON_COLORS } from '../../Buttons/IconButton';
+
+export default {
+	name: 'Modal',
+	components: { FeatureIcon, WnlButton, WnlIconButton },
+	props: {
+		size: {
+			type: String,
+			default: MODAL_SIZES.MEDIUM,
+			validator: (value: string) => Object.values(MODAL_SIZES).includes(value),
+		},
+		danger: {
+			type: Boolean,
+			default: false,
+		},
+		headerTitleSize: {
```

Comment:
HeaderSize?

Reply:
I updated Miro, it was confusing because not whole header was smaller, but only one element `title` so I renamed it. UX knows about it.